erpnext/selling/page/point_of_sale/pos_controller.js

```diff
@@ -637,11 +637,16 @@ erpnext.PointOfSale.Controller = class {
 		const bold_warehouse = warehouse.bold();
 		const bold_available_qty = available_qty.toString().bold()
 		if (!(available_qty > 0)) {
-			frappe.model.clear_doc(item_row.doctype, item_row.name);
-			frappe.throw({
-				title: __("Not Available"),
-				message: __('Item Code: {0} is not available under warehouse {1}.', [bold_item_code, bold_warehouse])
-			})
+			frappe.db.get_value('Item', item_row.item_code, 'is_stock_item').then(({message}) => {
```

Comment:
This `get_value` can be eliminated if you return it from whitelisted method `get_stock_availability`